### ISSUE TITLE
Spurious warning 'W: python-missing-require' when inspecting Fedora Python package

### ISSUE THREAD

**Issue by hroncok** [rpm-software-management/rpmlint#1171]
Downstream report https://bugzilla.redhat.com/show_bug.cgi?id=2260169

------

After updating to version 2.5.0 of rpmlint, I'm seeing false warning when inspecting Python RPM packages:

rpmlint: 2.5.0
configuration:
    /usr/lib/python3.12/site-packages/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora-legacy-licenses.toml
    /etc/xdg/rpmlint/fedora-spdx-licenses.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
checks: 32, packages: 2

fawltydeps.noarch: W: python-missing-require importlib_metadata
fawltydeps.noarch: W: python-missing-require isort
fawltydeps.noarch: W: python-missing-require pip-requirements-parser
fawltydeps.noarch: W: python-missing-require pydantic
fawltydeps.noarch: W: python-missing-require setuptools

Downgrading rpmlint to 2.4.0 makes this go away:

rpmlint: 2.4.0
configuration:
    /usr/lib/python3.12/site-packages/rpmlint/configdefaults.toml
    /etc/xdg/rpmlint/fedora-legacy-licenses.toml
    /etc/xdg/rpmlint/fedora-spdx-licenses.toml
    /etc/xdg/rpmlint/fedora.toml
    /etc/xdg/rpmlint/scoring.toml
    /etc/xdg/rpmlint/users-groups.toml
    /etc/xdg/rpmlint/warn-on-functions.toml
checks: 31, packages: 1

Reproducible: Always

Steps to Reproduce:
1. Upgrade rpmlint to 2.5.0
2. Grab an RPM from koji (e.g. https://kojipkgs.fedoraproject.org//packages/fawltydeps/0.13.3/1.fc40/noarch/fawltydeps-0.13.3-1.fc40.noarch.rpm)
3. Run rpmlint on the RPM
Actual Results:  
Spurious warning: 'W: python-missing-require'

Expected Results:  
No warning when there's no issue.

`rpm -q --requires` shows all dependencies as expected:

$ rpm -q --requires -p fawltydeps-0.13.3-1.fc40.noarch.rpm 
(python3.12dist(isort) < 6~~ with python3.12dist(isort) >= 5.10)
(python3.12dist(pip-requirements-parser) < 33~~ with python3.12dist(pip-requirements-parser) >= 32.0.1)
(python3.12dist(pydantic) < 3~~ with python3.12dist(pydantic) >= 1.10.2)
(python3.12dist(setuptools) < 69~~ with python3.12dist(setuptools) >= 65.5.1)
/usr/bin/python3
python(abi) = 3.12
python3.12dist(importlib-metadata) >= 6
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsZstd) <= 5.4.18-1
rpmlib(RichDependencies) <= 4.12.0-1
rpmlib(TildeInVersions) <= 4.10.0-1

------

I can personally reproduce this with e.g.

```
$ rpm -qR python3-jupyter-core
/usr/bin/python3
python(abi) = 3.12
python3.12dist(platformdirs) >= 2.5
python3.12dist(traitlets) >= 5.3
...

$ rpmlint --installed python3-jupyter-core
...
python3-jupyter-core.noarch: W: python-missing-require platformdirs
python3-jupyter-core.noarch: W: python-missing-require traitlets
...

```

**Comment by hroncok** [comment]
A very shallow inspection of this check makes me think:

 - in Fedora, we generate the dependencies from upstream metadata
 - rpmlint tries to compere manual requirements used in other distros against the same upstream metadata
 - rpmlint does not understand Fedora's generated dependencies and reports everything as missing
 
Possible solutions:
 - teach rpmlint to understand Fedora's requirements
 - disable this check in Fedora entirely, because it makes no sense there

**Comment by danigm** [comment]
> Possible solutions:
> 
>     * teach rpmlint to understand Fedora's requirements
>     * disable this check in Fedora entirely, because it makes no sense there

The python dependencies checker doesn't know about `python3.12dist(platformdirs)` kind of requirements. But if this check is not useful in fedora it can be disabled completely.

This check just tries to find missing requirements comparing the python package metadata and the requirements in the spec file, using some [name conversion and regex.](https://github.com/rpm-software-management/rpmlint/blob/main/rpmlint/checks/PythonCheck.py#L193-L198)

**Comment by hroncok** [comment]
> The python dependencies checker doesn't know about `python3.12dist(platformdirs)` kind of requirements.

Correct.

> But if this check is not useful in fedora it can be disabled completely.

It is not.

> This check just tries to find missing requirements comparing the python package metadata and the requirements in the spec file, using some [name conversion and regex.](https://github.com/rpm-software-management/rpmlint/blob/main/rpmlint/checks/PythonCheck.py#L193-L198)

The name conversion should probably be updated.
